File: VRP/VRPCore/src/inputTools/inputFile/City.cpp
#include "City.h"


City::City(int id, const std::string& name, const Point& point, int demand = 0, int depotId = 0, bool isFirst = false, bool isLast = false, bool isStop = false) :
	Depot::Depot(id, name, point),
	demand(demand),
	depotId(depotId),
	isFirst(isFirst),
	isLast(isLast),
	isStop(isStop)
{
}

City::City(const City& c) :
    Depot::Depot(c.id, c.name, c.point),
    demand(c.demand),
    depotId(c.depotId),
    isFirst(c.isFirst),
	isLast(c.isLast),
	isStop(c.isStop)
{
}

int City::getDemand() const
{
	return demand;
}

void City::setDemand(int demand)
{
	this->demand = demand;
}

int City::getDepotId() const
{
	return depotId;
}

void City::setDepotId(int depotId)
{
	this->depotId = depotId;
}

bool City::getIsFirst() const
{
	return isFirst;
}

void City::setIsFirst(bool isFirst)
{
	this->isFirst = isFirst;
}

bool City::getIsLast() const
{
	return isLast;
}

void City::setIsLast(bool isLast)
{
	this->isLast = isLast;
}

bool City::getIsStop() const
{
	return isStop;
}

void City::setIsStop(bool isStop)
{
	this->isStop = isStop;
}

bool City::isDepot() const
{
    return false;
}

File: JEE/src/main/java/vrp/jee/domain/City.java
package vrp.jee.domain;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlTransient;

/**
 *
 * @author Maciej Radzikowski <maciej@radzikowski.com.pl>
 */
public class City extends AbsLocation {

	@XmlTransient
	private Depot depot;

	private Integer demand = 0;

	private Boolean isFirst = false;

	private Boolean isLast = false;

	private Boolean isStop = false;

	@XmlElement
	public Integer getDepotId() {
		if (depot != null) {
			return depot.getId();
		}
		return null;
	}

	public Depot getDepot() {
		return depot;
	}

	public void setDepot(Depot depot) {
		this.depot = depot;
	}

	public Integer getDemand() {
		return demand;
	}

	public void setDemand(Integer demand) {
		this.demand = demand;
	}

	public Boolean getIsFirst() {
		return isFirst;
	}

	public void setIsFirst(Boolean isFirst) {
		this.isFirst = isFirst;
	}

	public Boolean getIsLast() {
		return isLast;
	}

	public void setIsLast(Boolean isLast) {
		this.isLast = isLast;
	}

	public Boolean getIsStop() {
		return isStop;
	}

	public void setIsStop(Boolean isStop) {
		this.isStop = isStop;
	}

}
